src/templates/plugin.routing.js

```diff
@@ -156,7 +156,7 @@ const VueInstanceProxy = function (targetFunction) {
   return function () {
     const proxy = {
       getRouteBaseName: this.getRouteBaseName,
-      i18n: this.$i18n,
+      i18n: this.$router.app.$i18n,
```

Comment:
This doesn't make sense to me. Why would we need to go through `$router` to get $i18n`. These two are unrelated.

And obviously `this.$i18n` is not undefined in general case or most of the tests would fail.
Please provide reproduction steps (or repo) so that I can check what is *actually* wrong there.

Reply:
We work with @freewhite4 on one project. Our nuxt.config.js is https://github.com/nuxt-community/nuxt-i18n/pull/630#issuecomment-598068821. We have a problem: switchLocalePath returns a link without a domain. @freewhite4 fixed the problem.

Reply:
I need to understand why it fixes the problem. This is surely not the way to fix it, whatever it is. Please create a minimal repo that reproduces.

Reply:
I can see how this change can make some sort of difference since `this.$router.app.$i18n` is a root `$i18n` instance (more direct path to it would be `this.$root.$i18n`) while `this.$i18n` can be a component-local instance if component defines some `i18n` options.

But it's not correct to use root instance always as it might not reflect the local state so this fix seems wrong.

The very latest version (https://github.com/nuxt-community/nuxt-i18n/releases/tag/v6.12.1) includes a fix for #557 that might actually fix your problem too. Have you tried that latest version?

Reply:
We used 6.11.1 version and had this problem. We migrated to 6.12.1 and the problem was resolved. 
Thank you!